sources/git.go

```diff
@@ -21,6 +30,27 @@ type GitCmd struct {
 	cmd         *exec.Cmd
 	diffFilesCh <-chan *gitdiff.File
 	errCh       <-chan error
+	repoPath    string
+}
+
+// blobReader provides a ReadCloser interface git cat-file blob to fetch
+// a blob from a repo
+type blobReader struct {
+	io.ReadCloser
+	cmd *exec.Cmd
+}
+
+// Close closes the underlying reader and then waits for the command to complete,
+// releasing its resources.
+func (br *blobReader) Close() error {
```

Comment:
This might need `io.Copy(io.Discard, br)` or it could lead to the process never exiting (https://github.com/trufflesecurity/trufflehog/pull/3375#issuecomment-2396822320). Worth double-checking.

Reply:
Good call :ok_hand: